client/src/components/Login/index.jsx

```diff
@@ -6,8 +6,31 @@ export default class extends Component {
   state = {
     email: '',
     password: '',
+    msg:''
   };
+  handelChange = ({ target: { name, value } }) => {
+    if(name ==="email"&& value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i))
+    this.setState({ [name]: value })
+    else this.setState({[name]:value})
```

Comment:
the ```if``` and the ```else``` statements should not do the same thing. It defeats the purpose of the ```if``` statement in the first place.

Reply:
if statement for check if the value of email input valid as email and the else for the password input value.

Reply:
@Jamalat-shamallakh In that case I think you might need `else if (name === "password") {`, because right now the `else` is handling both the password field *and* the case where it's the email field and it fails validation.

Another thing to consider is that this handleChange function will run on each keystroke, so the user's email might be considered invalid by the regex until they finish typing (this would be good to keep in mind if you were to show an error to the user if the regex fails)

Reply:
i will consider it